UnitsNet/GeneratedCode/Quantities/AmountOfSubstance.NetFramework.g.cs

```diff
@@ -753,7 +753,14 @@ public override int GetHashCode()
 
         #region Conversion Methods
 
-        double IQuantity.As(Enum unit) => As((AmountOfSubstanceUnit)unit);
+        /// <inheritdoc />
+        double IQuantity.As(Enum unit)
+        {
+            if(!(unit is AmountOfSubstanceUnit))
```

Comment:
Optional, but you can rewrite this to latest c# for extra vanity points:
```c#
if (unit is AmountOfSubstanceUnit validUnit)
{
    return As(validUnit);
}

throw new ArgumentException("The given unit is not of type AmountOfSubstanceUnit.", nameof(unit));
```

Reply:
Also, we could add `unit.GetType()` to the exception message to give extra help figuring out what went wrong.

Reply:
I only thought you could do that with ```as```. Sweet! And I agree, I updated the message.